math/kth/kthnumber.go

```diff
@@ -0,0 +1,47 @@
+package kth
+
+// FindKthMax use quick sort algorithm to get kth large element in given array
+// return -1 when not found
+func FindKthMax(nums []int, k int) int {
+	index := len(nums) - k
+	return kthNumber(nums, index)
+}
+
+// FindKthMin use quick sort algorithm to get kth small element in given array
+// return -1 when not found
+func FindKthMin(nums []int, k int) int {
+	index := k - 1
+	return kthNumber(nums, index)
+}
+
+func kthNumber(nums []int, k int) int {
+	if k < 0 || k >= len(nums) {
+		return -1
+	}
+	start := 0
+	end := len(nums)
+	for start < end {
+		pivot := partition(nums, start, end)
+		if k == pivot {
+			return nums[pivot]
+		} else if k > pivot {
+			start = pivot + 1
+		} else {
+			end = pivot
+		}
+	}
+	return -1
+}
+
+func partition(nums []int, start, end int) int {
```

Comment:
We already have a partition function, export it from the sort package and use that instead of creating a new one.